packages/dds/sequence/src/test/revertibles.spec.ts

```diff
@@ -90,11 +101,93 @@ describe("Sequence.Revertibles with Local Edits", () => {
 		const int = collection.getIntervalById(id);
 		assert.equal(int?.properties.foo, "one");
 	});
+	it("reverts multiple interval adds", () => {
+		collection.on("addInterval", (interval, local, op) => {
+			appendLocalAddToRevertibles(interval, revertibles);
+		});
+
+		sharedString.insertText(0, "hello world");
+		collection.add(0, 5, IntervalType.Simple);
+		collection.add(5, 7, IntervalType.Simple);
+
+		revertIntervalRevertibles(sharedString, revertibles.splice(0));
+		assertIntervals(sharedString, collection, []);
+	});
+	it("reverts multiple interval removes", () => {
+		collection.on("deleteInterval", (interval, local, op) => {
+			appendLocalDeleteToRevertibles(sharedString, interval, revertibles);
+		});
+
+		sharedString.insertText(0, "hello world");
+		// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+		const id = collection.add(0, 5, IntervalType.Simple).getIntervalId()!;
+		// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+		const id2 = collection.add(5, 7, IntervalType.Simple).getIntervalId()!;
+		collection.removeIntervalById(id);
+		collection.removeIntervalById(id2);
+
+		revertIntervalRevertibles(sharedString, revertibles.splice(0));
+		assertIntervals(sharedString, collection, [
+			{ start: 0, end: 5 },
+			{ start: 5, end: 7 },
+		]);
+	});
+	it("performs multiple reverts on the same interval", () => {
+		collection.on("addInterval", (interval, local, op) => {
+			appendLocalAddToRevertibles(interval, revertibles);
+		});
+
+		collection.on("deleteInterval", (interval, local, op) => {
+			appendLocalDeleteToRevertibles(sharedString, interval, revertibles);
+		});
+
+		collection.on("changeInterval", (interval, previousInterval, local, op) => {
+			appendLocalChangeToRevertibles(sharedString, interval, previousInterval, revertibles);
+		});
+
+		sharedString.insertText(0, "hello world");
+		// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+		const id = collection.add(0, 5, IntervalType.Simple).getIntervalId()!;
+		collection.change(id, 3, 8);
+		collection.removeIntervalById(id);
+
+		revertIntervalRevertibles(sharedString, revertibles.splice(0));
+		assertIntervals(sharedString, collection, []);
+	});
+	it("performs two local changes, then reverts the first", () => {
```

Comment:
This is actually reverting both the add and change, which doesn't match the description.

Also, it's not really supported to revert local edits out of order, although I think it would work fine with your code.

Reply:
Never mind. There's no revertible for the change since only add is subscribed to.